src/tests/ftest/control/ms_resilience.py

```diff
@@ -0,0 +1,244 @@
+    """
+
+    def __init__(self, *args, **kwargs):
+        """Inititialize a ManagementServiceResilience object."""
+        super(ManagementServiceResilience, self).__init__(*args, **kwargs)
+        self.setup_start_servers = False
+
+    def update_and_verify(self, ignore_status=False):
+        """Create a pool on the server group
+
+        Args:
+            ignore_status (bool): If False, raise an exception when pool create
+                fails, otherwise don't raise an exception. Defaults to False.
+        """
+        self.add_pool(create=False)
+        self.pool.name.value = self.server_group
+        try:
+            self.pool.create()
+        except TestFail as error:
+            if ignore_status:
+                self.log.info("Expected MS error creating pool!: %s", error)
+            else:
+                self.fail("Pool create failed unexpectedly!")
+
+        if not ignore_status:
+            self.log.info("Pool UUID %s on server group: %s",
+                          self.pool.uuid, self.server_group)
+            # Verify that the pool persisted.
+            pool_data = self.server_managers[-1].dmg.pool_list()
+            if pool_data and self.pool.uuid in pool_data:
+                self.log.info("Found pool in system.")
+            else:
+                self.fail("No pool found in system.")
+
+    def verify_leader(self, access_list):
+        """Verify the leader of the MS is in the access_list.
+
+        Args:
+            access_list (list): list of hostname representing the access points
+                for the MS.
+
+        Returns:
+            str: hostname of the MS leader. If the leader is not within the
+                access_list, the test will fail.
+
+        """
+        l_addr = self.server_managers[-1].dmg.system_leader_query()["leader"]
+        l_hostname, _, _ = socket.gethostbyaddr(l_addr.split(":")[0])
+        l_hostname = l_hostname.split(".")[0]
+
+        # Check that the new leader is in the access list
+        if l_hostname not in access_list:
+            self.fail("Selected leader <{}> is not within the access_list"
+                      " provided to servers".format(l_hostname))
+        return l_hostname
+
+    def launch_servers(self, resilience_num):
+        """Setup and start the daos_servers.
+
+        Args:
+            resilience_num (int): minimum amount of MS replicas to achieve
+                resiliency.
+
+        Returns:
+            list: list of access point hosts where MS has been started.
+
+        """
+        access_list = random.sample(self.hostlist_servers, resilience_num)
+        server_groups = {
+            self.server_group:
+                {
+                    "servers": self.hostlist_servers,
+                    "access_list": access_list
+                },
+        }
+        self.start_servers(server_groups)
+        return access_list
+
+    def verify_resiliency(self, N):
+        """Verify 2N+1 resiliency up to N = 2"""
```

Comment:
It would be great if you could add the full method description. Do we support only up to 2 for N?
Also, it looks like only the leader is killed in this method. Are we testing the case where we verify resiliency after killing multiple (N) hosts?

Reply:
Yes, will add. Currently the max number for N that could be supported is N=3. we could add the remaining variants to the test, but I decided to do 2 to keep the test time as fasts as possible while testing what we need at the same time.

And yes, we are testing both cases on `verify_resiliency`  and `verify_lost_resiliency`